cpp/src/lexer/lexer.hh

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <string_view>
+
+enum class TokenType
+{
+	illegal,
+	eof,
+	identifier,
+	integer,
+	assign,
+	plus,
+	comma,
+	semicolon,
+	lparen,
+	rparen,
+	lsquirly,
+	rsquirly,
+	function,
+	let,
+};
+
+
+struct Token final
+{
+	const TokenType type;
```

Comment:
This const seems too harsh. It effectively strips away copy-assignability. As a user, I'd expect the Token returned by nextToken() to be a trivially copyable simple struct. Let me take a const Token if I mean it.